lib/dns/resource.js

```diff
@@ -1224,13 +1228,27 @@ class Resource extends Struct {
         key.signZSK(res.additional, types.A);
         key.signZSK(res.additional, types.AAAA);
         break;
-      case types.NS:
+      case types.NS: {
         res.authority = this.toNS(name);
         res.additional = this.toNSIP(name);
         for (const rr of this.toGlue(types.NS))
           res.additional.push(rr);
         key.signZSK(res.authority, types.NS);
+
```

Comment:
This logic handles the case when there are multiple names of the same RR type. I think this logic should move inside of `key.signZSK` so that the abstraction here can stay the same. A single `key.signZSK` would be able to append multiple `RRSIG` records if there are multiple rrsets in the same section, ie they have different names but the same rr type (since that is the second argument to `key.signZSK`.

Reply:
This is probably good to note for the future, but is no longer relevant.